src/ProjectorMonteCarloProblem.jl

```diff
@@ -0,0 +1,261 @@
+"""
+    FCIQMC()
+Algorithm for the full configuration interaction quantum Monte Carlo (FCIQMC) method.
+To be passed as the `algorithm` keyword argument to [`ProjectorMonteCarloProblem`](@ref).
+"""
+struct FCIQMC end
+
+"""
+    SimulationPlan(; starting_step = 1, last_step = 100, walltime = Inf)
+Defines the duration of the simulation. The simulation ends when the `last_step` is reached
+or the `walltime` is exceeded.
+
+See [`ProjectorMonteCarloProblem`](@ref), [`PMCSimulation`](@ref).
+"""
+Base.@kwdef struct SimulationPlan
+    starting_step::Int = 0
+    last_step::Int = 100
+    walltime::Float64 = Inf
+end
+
```

Comment:
A `SimulationPlan` currently prints as
```
SimulationPlan(0, 100, Inf)
```
It would be better if the show method displayed the argument names as well.

```suggestion
function Base.show(io::IO, plan::SimulationPlan)
    print(
        io, "SimulationPlan(starting_step=", plan.starting_step,
        ", last_step=", plan.last_step, ", walltime=", plan.walltime, ")"
    )
end
```